### ISSUE TITLE
Crash with view for vertex table 

### ISSUE THREAD

**Issue by eaubin** [cwida/duckpgq-extension#158]
### What happens?

Hi, I'm trying the extension for the first time and ran into a crash with:

```sql
CREATE TABLE vdata(id TEXT PRIMARY KEY);
CREATE VIEW v AS SELECT * FROM vdata;
CREATE TABLE w(id TEXT PRIMARY KEY);
CREATE TABLE v_w(v_id TEXT, w_id TEXT);
LOAD duckpgq;
CREATE PROPERTY GRAPH g1
  VERTEX TABLES (v, w)

  EDGE TABLES (
    v_w SOURCE KEY (v_id) REFERENCES v (id)
        DESTINATION KEY (w_id) REFERENCES w (id)
    LABEL hasW
  );
```

it crashes at

```
thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
  * frame #0: 0x0000000000000000
    frame #1: 0x00000001038af5e0 duckpgq.duckdb_extension`duckdb::Binder::BindCreateViewInfo(duckdb::CreateViewInfo&) + 224
    frame #2: 0x00000001038b3bb8 duckpgq.duckdb_extension`duckdb::Binder::Bind(duckdb::CreateStatement&) + 1676
    frame #3: 0x00000001037f4078 duckpgq.duckdb_extension`duckpgq::core::duckpgq_bind(duckdb::ClientContext&, duckdb::Binder&, duckdb::OperatorExtensionInfo*, duckdb::SQLStatement&) + 216
    frame #4: 0x000000010017b930 duckdb`duckdb::Planner::CreatePlan(duckdb::SQLStatement&) + 1188
    frame #5: 0x0000000100d0d6c4 duckdb`duckdb::ClientContext::CreatePreparedStatementInternal(duckdb::ClientContextLock&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, duckdb::unique_ptr<duckdb::SQLStatement, std::__1::default_delete<duckdb::SQLStatement>, true>, duckdb::optional_ptr<std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, duckdb::BoundParameterData, duckdb::CaseInsensitiveStringHashFunction, duckdb::CaseInsensitiveStringEquality, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const, duckdb::BoundParameterData>>>, true>) + 512
    frame #6: 0x0000000100d0e0c4 duckdb`duckdb::ClientContext::CreatePreparedStatement(duckdb::ClientContextLock&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, duckdb::unique_ptr<duckdb::SQLStatement, std::__1::default_delete<duckdb::SQLStatement>, true>, duckdb::optional_ptr<std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, duckdb::BoundParameterData, duckdb::CaseInsensitiveStringHashFunction, duckdb::CaseInsensitiveStringEquality, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const, duckdb::BoundParameterData>>>, true>, duckdb::PreparedStatementMode) + 480
    frame #7: 0x0000000100d6574c duckdb`std::__1::__function::__func<duckdb::ClientContext::PrepareInternal(duckdb::ClientContextLock&, duckdb::unique_ptr<duckdb::SQLStatement, std::__1::default_delete<duckdb::SQLStatement>, true>)::$_2, std::__1::allocator<duckdb::ClientContext::PrepareInternal(duckdb::ClientContextLock&, duckdb::unique_ptr<duckdb::SQLStatement, std::__1::default_delete<duckdb::SQLStatement>, true>)::$_2>, void ()>::operator()() + 60
    frame #8: 0x0000000100d0ffb8 duckdb`duckdb::ClientContext::RunFunctionInTransactionInternal(duckdb::ClientContextLock&, std::__1::function<void ()> const&, bool) + 124
    frame #9: 0x0000000100d102ac duckdb`duckdb::ClientContext::PrepareInternal(duckdb::ClientContextLock&, duckdb::unique_ptr<duckdb::SQLStatement, std::__1::default_delete<duckdb::SQLStatement>, true>) + 216
    frame #10: 0x0000000100d10658 duckdb`duckdb::ClientContext::Prepare(duckdb::unique_ptr<duckdb::SQLStatement, std::__1::default_delete<duckdb::SQLStatement>, true>) + 168
    frame #11: 0x0000000100d1a35c duckdb`duckdb::Connection::Prepare(duckdb::unique_ptr<duckdb::SQLStatement, std::__1::default_delete<duckdb::SQLStatement>, true>) + 52
    frame #12: 0x0000000100028830 duckdb`duckdb_shell_sqlite3_prepare_v2 + 788
    frame #13: 0x000000010000eb54 duckdb`shell_exec + 184
    frame #14: 0x000000010001a258 duckdb`runOneSqlLine + 156
    frame #15: 0x000000010000f9c4 duckdb`process_input + 964
    frame #16: 0x00000001000082b8 duckdb`main + 1788
    frame #17: 0x0000000181063154 dyld`start + 2476
```

In my real application I use a lot of views for what would be both the edge tables and the vertex tables. Would pgq still be useful without all the foreign keys?

### To Reproduce

```sql
CREATE TABLE vdata(id TEXT PRIMARY KEY);
CREATE VIEW v AS SELECT * FROM vdata;
CREATE TABLE w(id TEXT PRIMARY KEY);
CREATE TABLE v_w(v_id TEXT, w_id TEXT);
LOAD duckpgq;
CREATE PROPERTY GRAPH g1
  VERTEX TABLES (v, w)

  EDGE TABLES (
    v_w SOURCE KEY (v_id) REFERENCES v (id)
        DESTINATION KEY (w_id) REFERENCES w (id)
    LABEL hasW
  );
```

### OS:

osx, arm64

### DuckDB Version:

1.1.2

### DuckDB Client:

unix cmd line

### Full Name:

eaubin

### Affiliation:

n/a

### How did you load the extension?

LOAD duckpgq;

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by Dtenwolde** [comment]
Thanks for the report! Unfortunately, I don't have an immediate solution to support property graph creation on views. Internally they are different than standard tables. I just pushed b29e8ab27d6aae8e61eaee991ba41b5a269cb6fc and ac29e245d2172eb8f2e696e2f1ae0708c3a34bd6 that throws a better error message. In your case: 
`Invalid Error: Found a view with name v. Creating property graph tables over views is currently not supported.`

> Would pgq still be useful without all the foreign keys?
I am not exactly sure what you mean with foreign keys in this case, sorry. To support views for property graphs I need more time to figure that out, but I wanted to at least throw a proper error message instead of crashing. 

In the mean time I have created this issue to track the progress: https://github.com/cwida/duckpgq-extension/issues/159
